src/Silicon.cpp

```diff
@@ -0,0 +1,506 @@
+            double y_off = y - yoff[n];
+            if (silicon.insidePixel(ix_off, iy_off, x_off, y_off, zconv, target)) {
+                xdbg<<"Found in pixel "<<n<<", ix = "<<ix<<", iy = "<<iy
+                    <<", x="<<x<<", y = "<<y<<", target(ix,iy)="<<target(ix,iy)<<std::endl;
+                ix = ix_off;
+                iy = iy_off;
+                return true;
+            }
+            n = ((n-1) + step) % 8 + 1;
+            // This is intended to start with the nearest neighbor, then cycle through others.
+        }
+        return false;
+    }
+
+    template <typename T>
+    double Silicon::accumulate(const PhotonArray& photons, UniformDeviate ud, ImageView<T> target)
+    {
+        Bounds<int> b = target.getBounds();
+        if (!b.isDefined())
+            throw std::runtime_error("Attempting to PhotonArray::addTo an Image with"
+                                     " undefined Bounds");
+
+        // Factor to turn flux into surface brightness in an Image pixel
+#ifdef DEBUGLOGGING
+        dbg<<"In Silicon::accumulate\n";
+        dbg<<"bounds = "<<b<<std::endl;
+        dbg<<"total nphotons = "<<photons.size()<<std::endl;
+        dbg<<"hasAllocatedWavelengths = "<<photons.hasAllocatedWavelengths()<<std::endl;
+        dbg<<"hasAllocatedAngles = "<<photons.hasAllocatedAngles()<<std::endl;
+        double Irr = 0.;
+        double Irr0 = 0.;
+        int zerocount=0, neighborcount=0, misscount=0;
+#endif
+
+        int nx = b.getXMax() - b.getXMin() + 1;
+        int ny = b.getYMax() - b.getYMin() + 1;
+        dbg<<"nx,ny = "<<nx<<','<<ny<<std::endl;
+        _imagepolys.resize(nx*ny);
+        for (int i=0; i<(nx*ny); ++i)
+            _imagepolys[i] = _emptypoly;
+        dbg<<"Built poly list\n";
+
+        const int nphotons = photons.size();
+        std::vector<double> depth(nphotons);
+        calculateConversionDepth(photons, depth, ud);
+
+        GaussianDeviate gd(ud,0,1); // Random variable from Standard Normal dist.
+
+        double addedFlux = 0.;
+        double next_recalc = _nrecalc;
+        for (int i=0; i<nphotons; i++) {
+            // Update shapes every _nrecalc electrons
+            if (addedFlux > next_recalc) {
+                updatePixelDistortions(target);
+                next_recalc = addedFlux + _nrecalc;
+            }
+
+            // Get the location where the photon strikes the silicon:
+            double x0 = photons.getX(i); // in pixels
+            double y0 = photons.getY(i); // in pixels
+
+            double zconv;  // This is the reverse of depth. zconv is how far above the substrate.
+            if (photons.hasAllocatedAngles()) {
+                double dxdz = photons.getDXDZ(i);
+                double dydz = photons.getDYDZ(i);
+                double dz = depth[i];
+                x0 += dxdz * dz / 10.0; // in pixels
+                y0 += dydz * dz / 10.0; // in pixels
+                zconv = 100.0 - dz; // Conversion depth in microns
```

Comment:
I assume from this (and the 95 above) that the height of the silicon is 100 microns.  I think this should also be an input parameter, but I couldn't find it in you config file.  Isn't this something you need to specify for your Poisson model?  Or do you just have it hard-coded to 100 microns, rather than using a variable?

I note here that the calculation of `diff_step` in the Python layer also hard codes the thickness as 100 microns.